### PR TITLE
feat: Add user agent for checks

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Matches what we did over here https://github.com/getsentry/sentry/pull/75673

### CODE REVIEW

File: src/checker/http_checker.rs [hunk 1]
```diff
@@ -63,7 +67,11 @@ fn dns_error(err: &reqwest::Error) -> Option<String> {
 
 impl HttpChecker {
     pub fn new() -> Self {
+        let mut default_headers = HeaderMap::new();
+        default_headers.insert("User-Agent", UPTIME_USER_AGENT.to_string().parse().unwrap());
```
Comment (wedamija) [inline src/checker/http_checker.rs:71]: Shouldn't we be using the agent from the config?

Comment (evanpurkhiser) [reply src/checker/http_checker.rs:71]: we don't have it in the config yet do we?

Comment (evanpurkhiser) [reply src/checker/http_checker.rs:71]: We can slap it in there if we want

Comment (wedamija) [reply src/checker/http_checker.rs:71]: Oh right, yeah we can do that whenever then. It's more important for when people are able to configure their own uptime monitors

**Review by wedamija — APPROVED** [review]

**Pull request merged**